test/fixtures/roles.sql

```diff
@@ -5,3 +5,32 @@ CREATE ROLE postgrest_test_default_role;
 CREATE ROLE postgrest_test_author;
 
 GRANT postgrest_test_anonymous, postgrest_test_default_role, postgrest_test_author TO :USER;
+
+-- reloadable config options for io tests
+ALTER ROLE postgrest_test_authenticator SET pgrst."jwt-aud" = 'https://example.org';
+ALTER ROLE postgrest_test_authenticator SET pgrst."openapi-server-proxy-uri" = 'https://example.org/api';
```

Comment:
while naming the settings like `pgrst."openapi-server-proxy-uri"`, the alter statement works, and they are stored in the pg config tables, however, all the settings named like this can not be read using `show setting` or `current_setting`, and it should work under normal conditions (i bet that is why you used that semi complicated statement to get them back).

The problem here is the `-` character, it it were named like `pgrst.openapi_server_proxy_uri` (or maybe `PGRST.OPENAPI_SERVER_PROXY_URI` like the env vars) then `show` and `current_setting` functions work properly and the `dbSettingsStatement` statement can be a bit more clear/explicit as to what it's fetching from the database

Reply:
discussion here https://www.postgresql.org/message-id/flat/20210209144059.GA21360%40depesz.com

Reply:
It might not work with `SHOW`, but it does work with `current_setting`.

```sql
set app."test-a-b-c" = 'hello world';
select current_setting('app.test-a-b-c');
```

returns "hello world" for me. No problem there.

Reply:
PG 13.1 here.

Reply:
Hm. I'm probably mis-understanding something. The same thing works for me with `SHOW`.

Reply:
Hm. Happens only when using `ALTER ROLE ... SET` - then loading the guc on login is broken once there is a `-` in it. I'd consider this a bug in Postgres.

Reply:
Ah, can reproduce as well on pg 11.5 - not even `current_setting` works when the guc has a dash. I guess this should be fine for our config since we get it through `pg_db_role_setting`. But not for using the guc configs through functions, which should be the case for `jwt-secret`.

Not sure if we should change the dashes to underscores, looks like this is already been considered as a bug on the pg mailing list.

Reply:
Looking at the latest post on the mailing list, the bug seems to be that those variables are actually available with `_` instead of `-` already.

So for all PG versions that have the bug, `SELECT current_setting('pgrst.openapi_server_proxy_uri')` should work even without changes from our side.

Reply:
@wolfgangwalther Great! I've also confirmed this works. 

Though I guess it'd be good to keep the config option names consistent. I'll change the dashes to underscores on another PR.

Reply:
This will be our only way forward, too, because the consensus on the PG mailing list now is to don't allow `-` in GUC names anymore.